### ISSUE TITLE
Change Prepare > Column: Text > Find/Replace dialogue again!

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#7120]
Apologies @N-thony but, just after you agreed to make a new issue of a minor change in this dialogue, I am requesting larger changes.  Please bear with me, because I think this is a clear improvement.  I hope it can be quick, as it is almost all cosmetic!  These have become obvious as I am now trying to write the help.  I am getting to something simpler.

a) The menu says Find/Replace, so the heading in the dialogue, which is currently String Handling could also change to become Find/Replace.
b) Then the idea of the top radio buttons is that they correspond to different layouts in the dialogue.  This isn't the case here.  So please change them to just 2 (familiar) radio buttons, namely just `Find` and `Replace`.
c) So, the dialogue will change in shape a bit.  It can be a little less wide and a little deeper.  The 2 (or 3) existing controls could perhaps be raised a little.
d) Then Find will have a group box under the 2 existing controls.  It might have the label Options.  I would like a better word.  Inside are (at least) Count, Detect, Locate, Extract.  
e) The second has just Replace and Replace All.
f) The names for the resulting variables can remain as they are.

**Comment by N-thony** [comment]
> Apologies @N-thony but, just after you agreed to make a new issue of a minor change in this dialogue, I am requesting larger changes. Please bear with me, because I think this is a clear improvement. I hope it can be quick, as it is almost all cosmetic! These have become obvious as I am now trying to write the help. I am getting to something simpler.
> 
> a) The menu says Find/Replace, so the heading in the dialogue, which is currently String Handling could also change to become Find/Replace. b) Then the idea of the top radio buttons is that they correspond to different layouts in the dialogue. This isn't the case here. So please change them to just 2 (familiar) radio buttons, namely just `Find` and `Replace`. c) So, the dialogue will change in shape a bit. It can be a little less wide and a little deeper. The 2 (or 3) existing controls could perhaps be raised a little. d) Then Find will have a group box under the 2 existing controls. It might have the label Options. I would like a better word. Inside are (at least) Count, Detect, Locate, Extract. e) The second has just Replace and Replace All. f) The names for the resulting variables can remain as they are.

@rdstern no problem, still happy to work on this. Would you prefer me to work on this after merging PR #7118?

**Comment by rdstern** [comment]
Yes, please wait.  I am also still working out details of the new design

**Comment by rdstern** [comment]
So here it is.  Not as different as I thought before, but adding useful parts of stringr, just as you included those from forcats when improving the factor dialogues.
a) The title becomes Find/Replace
b) There are 5 buttons at the top, namely Detect, Find, Replace, Replace NA, Remove
c) In `Detect `use the existing Detect.  Under the 2 existing controls there are 3 radio buttons, labelled Detect, Starts, Ends.  There is also a checkbox, Negate, default unchecked.  (It applies to any of these 3 options.  The saving name can remain as detect for all options.
d) In `Find` start with any of the existing options, for Count, Extract, Locate.  Add radio buttons under the controls with options Count, Extract, Locate, (and new one) Match.  Add a checkbox All, default unchecked.  Ideally this is not visible for the Count option.  The suggested column names would ideally depend on the option selected, i.e. count, extract, locate, match.
e) There is (I think) an additional complication for Locate, which generates 2 variables.  Possibly leave that for now, but consider later.
f) `Replace `is as now, but with a checkbox All.  (So it would cover the 2 current options Replace and Replace All.)
g) `Replace NA` can be an adaptation of any of the existing tabs.  It has the single receiver.  The Pattern control is replaced by a checkbox with label `Replace by:`.  Default is unchecked.   If checked, then it opens a Pattern field with NA as the default.  You can type anything.  The Regex fields are not here or at least not visible.
h) `Remove` has the same single receiver and Pattern as the Find options.  It doesn't have the Replace by, because it is removing the pattern.  It does have the All checkbox, default unchecked.
